src/System.Net.Http/src/System/Net/Http/Managed/ConnectionCloseStream.cs

```diff
@@ -40,24 +38,34 @@ public ConnectionCloseReadStream(HttpConnection connection)
                 return bytesRead;
             }
 
+            public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
```

Comment:
Doesn't this mean that we will allow writes on any Connection: close connection?  This seems like it's too general.  Can't we restrict this to cases where we are actually using the Upgrade mechanism?

Reply:
> Doesn't this mean that we will allow writes on any Connection: close connection?

Yes

> Can't we restrict this to cases where we are actually using the Upgrade mechanism?

I guess?  So you want to check whether there was an Upgrade request and Upgrade response header and allow writes only when both are there?

Reply:
I assume we'd look for a 101 response code and then do something special with it (i.e. give back the duplex stream).

Reply:
Hmm. Ok.

Reply:
This does seem useful in other cases than WebSocket, btw.  

Basically we are saying that we support the Upgrade mechanism, and we have a way to expose the resulting raw stream back to the user.

This also gets us part way to supporting CONNECT tunneling for proxies...

Reply:
> This does seem useful in other cases than WebSocket, btw.

Yeah, that's why I didn't constrain it.  But I can see the argument in constraining it only to a switching protocols response code, so I'll do so.